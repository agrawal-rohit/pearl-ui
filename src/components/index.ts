// Atoms
export { default as Box, BoxProps } from "./atoms/box/box";
export { default as Center } from "./atoms/center/center";
export { default as Collapse, CollapseProps } from "./atoms/collapse/collapse";
export { default as Divider, DividerProps } from "./atoms/divider/divider";
export { default as Fade, FadeProps } from "./atoms/fade/fade";
export { default as Grid, GridProps } from "./atoms/grid/grid";
export { default as Icon, IconProps } from "./atoms/icon/icon";
export {
  default as Pressable,
  PressableProps,
} from "./atoms/pressable/pressable";
export {
  default as ScaleFade,
  ScaleFadeProps,
} from "./atoms/scale-fade/scale-fade";
export {
  default as List,
  OrderedList,
  UnorderedList,
  ListProps,
} from "./atoms/list/list";
export { default as Screen, ScreenProps } from "./atoms/screen/screen";
export { default as Skeleton, SkeletonProps } from "./atoms/skeleton/skeleton";
export {
  default as SkeletonCircle,
  SkeletonCircleProps,
} from "./atoms/skeleton/skeleton-circle";
export {
  default as SkeletonText,
  SkeletonTextProps,
} from "./atoms/skeleton/skeleton-text";
export { default as Slide, SlideProps } from "./atoms/slide/slide";
export {
  default as SlideFade,
  SlideFadeProps,
} from "./atoms/slide-fade/slide-fade";
export { default as Spacer } from "./atoms/spacer/spacer";
export { default as Spinner, SpinnerProps } from "./atoms/spinner/spinner";
export {
  default as Stack,
  HStack,
  VStack,
  ZStack,
  HStackProps,
  VStackProps,
  StackProps,
} from "./atoms/stack/stack";
export { default as Text, TextProps } from "./atoms/text/text";

// Molecules
export { default as Avatar, AvatarProps } from "./molecules/avatar/avatar";
export {
  default as AvatarGroup,
  AvatarGroupProps,
} from "./molecules/avatar/avatar-group";
export { default as Badge, BadgeProps } from "./molecules/badge/badge";
export {
  default as withBadge,
  WithBadgeOptions,
} from "./molecules/badge/withBadge";
export { default as Button, ButtonProps } from "./molecules/button/button";
export {
  default as ButtonGroup,
  ButtonGroupProps,
} from "./molecules/button/button-group";
export {
  default as CheckBox,
  CheckBoxProps,
} from "./molecules/checkbox/checkbox";
export {
  default as CheckBoxGroup,
  CheckBoxGroupProps,
} from "./molecules/checkbox/checkbox-group";
export {
  default as IconButton,
  IconButtonProps,
} from "./molecules/icon-button/icon-button";
export { default as Image, ImageProps } from "./molecules/image/image";
export { default as Input, InputProps } from "./molecules/input/input";
export {
  default as PinInput,
  PinInputProps,
} from "./molecules/pin-input/pin-input";
export {
  default as Progress,
  ProgressProps,
} from "./molecules/progress/progress";
export { default as Radio, RadioProps } from "./molecules/radio/radio";
export {
  default as RadioGroup,
  RadioGroupProps,
} from "./molecules/radio/radio-group";
export { default as Switch, SwitchProps } from "./molecules/switch/switch";
export {
  default as Textarea,
  TextareaProps,
} from "./molecules/textarea/textarea";
export { default as Video, VideoProps } from "./molecules/video/video";
export {
  default as Accordion,
  AccordionProps,
  AccordionItemProps,
  AccordionButtonProps,
  AccordionPanelProps,
  AccordionItem,
  AccordionButton,
  AccordionPanel,
} from "./molecules/accordion/accordion";
export { default as Alert, AlertProps } from "./molecules/alert/alert";
