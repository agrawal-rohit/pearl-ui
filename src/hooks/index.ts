export { useColorModeValue } from "./useColorModeValue";
export { useComponentConfig } from "./useComponentConfig";
export { useStyledProps } from "./useStyledProps";
export { useTheme } from "./useTheme";
