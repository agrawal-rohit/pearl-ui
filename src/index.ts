// Hooks
export * from "./hooks/index";

// Components
export * from "./components/index";

// Animations
export * from "./animations";

// Theme
export * from "./theme/index";

// Utils
export * from "./utils/index";

// pearl
export * from "./pearl";
