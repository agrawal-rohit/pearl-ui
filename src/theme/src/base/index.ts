import { breakpoints } from "./breakpoints";
import { BasePearlTheme, FinalPearlTheme } from "../types";

import { borderRadii } from "./borderRadii";
import { elevation } from "./elevation";
import { palette } from "./palette";
import { spacing } from "./spacing";
import { zIndices } from "./zIndices";
import {
  fontConfig,
  fonts,
  fontSizes,
  fontWeights,
  lineHeights,
} from "./typography";

// Component configs
import SpinnerConfig from "../../../components/Atoms/Spinner/Spinner.config";
import ScreenConfig from "../../../components/Atoms/Screen/Screen.config";
import TextConfig from "../../../components/Atoms/Text/Text.config";
import ButtonConfig from "../../../components/Molecules/Button/Button.config";
import IconConfig from "../../../components/Atoms/Icon/Icon.config";
import TextLinkConfig from "../../../components/Molecules/TextLink/TextLink.config";
import InputConfig from "../../../components/Molecules/Input/Input.config";
import DividerConfig from "../../../components/Atoms/Divider/Divider.config";
import CheckBoxConfig from "../../../components/Molecules/CheckBox/CheckBox.config";
import ImageConfig from "../../../components/Molecules/Image/Image.config";
import RadioConfig from "../../../components/Molecules/Radio/Radio.config";
import BadgeConfig from "../../../components/Molecules/Badge/Badge.config";
import AvatarConfig from "../../../components/Molecules/Avatar/Avatar.config";
import { isFunction, mergeWith } from "lodash";

export const baseTheme = {
  palette,
  spacing,
  borderRadii,
  breakpoints,
  elevation,
  zIndices,
  fonts,
  fontConfig,
  fontSizes,
  fontWeights,
  lineHeights,
  components: {
    Icon: IconConfig,
    Text: TextConfig,
    Screen: ScreenConfig,
    Spinner: SpinnerConfig,
    Button: ButtonConfig,
    TextLink: TextLinkConfig,
    Input: InputConfig,
    CheckBox: CheckBoxConfig,
    Radio: RadioConfig,
    Badge: BadgeConfig,
    Divider: DividerConfig,
    Image: ImageConfig,
    Avatar: AvatarConfig,
  },
};

/**
 * Override particular parts of the default theme to create a custom theme as per you app's needs
 * @param overrideConfig Configuration overrides to be combined with the default theme
 * @returns
 */
export const extendTheme = <
  T extends typeof baseTheme | (Record<string, any> & {})
>(
  overrideConfig: T
) => {
  function customizer(source: any, override: any) {
    if (isFunction(source)) {
      return (...args: any[]) => {
        const sourceValue = source(...args);
        const overrideValue = isFunction(override)
          ? override(...args)
          : override;
        return mergeWith({}, sourceValue, overrideValue, customizer);
      };
    }
    return undefined;
  }

  const finalTheme = [overrideConfig].reduce((prevValue, currentValue) => {
    return mergeWith({}, prevValue, currentValue, customizer);
  }, baseTheme);

  return finalTheme as T & typeof baseTheme;
};
