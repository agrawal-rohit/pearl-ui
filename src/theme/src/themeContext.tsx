import AppLoading from "expo-app-loading";
import {
  useFonts,
  Poppins_300Light,
  Poppins_400Regular,
  Poppins_500Medium,
  Poppins_600SemiBold,
  Poppins_700Bold,
  Poppins_800ExtraBold,
} from "@expo-google-fonts/poppins";
import React, { useEffect, useState, createContext } from "react";
import { useColorScheme } from "react-native";
import { baseTheme } from "./basetheme";
import { BasePearlTheme } from "./types";

type ThemeType = "light" | "dark" | "system";

export interface IThemeContext {
  /** Theme configuration object for the active color mode */
  theme: BasePearlTheme;
  /** Active color mode */
  colorMode: ThemeType;
  /** Function to toggle the active color mode */
  toggleColorMode(): void;
}

interface ThemeProviderProps {
  /** Initial color mode for the theme (light, dark, system) */
  initialColorMode?: ThemeType;
  /** The theme configuration object */
  theme?: BasePearlTheme;
  /** A mapping object of the fonts used by the theme */
  fonts?: object;
  /**React children */
  children: React.ReactNode;
}

const baseFonts = {
  "Poppins-Light": Poppins_300Light,
  "Poppins-Regular": Poppins_400Regular,
  "Poppins-Medium": Poppins_500Medium,
  "Poppins-SemiBold": Poppins_600SemiBold,
  "Poppins-Bold": Poppins_700Bold,
  "Poppins-ExtraBold": Poppins_800ExtraBold,
};

export const themeContext = createContext({} as IThemeContext);

/**
 * The main provider component which controls the theme and color mode of the components in the entire application.
 */
export const ThemeProvider: React.FC<ThemeProviderProps> = ({
  initialColorMode = "light",
  theme = baseTheme,
  fonts = baseFonts,
  children,
}) => {
  const [colorMode, setColorMode] = useState<ThemeType>(initialColorMode);
  const systemThemeStyle = useColorScheme() as ThemeType;
  const [haveFontsLoaded] = useFonts(fonts);

  const changeThemeTo = (colorMode: ThemeType) => {
    if (colorMode === "dark") {
      setColorMode("dark");
    } else {
      setColorMode("light");
    }
  };

  const toggleColorMode = () => {
    if (colorMode === "light") {
      changeThemeTo("dark");
    } else {
      changeThemeTo("light");
    }
  };

  useEffect(() => {
    // Set initial Theme
    if (colorMode === "light") {
      changeThemeTo("light");
    } else if (colorMode === "dark") {
      changeThemeTo("dark");
    } else if (colorMode === "system") {
      changeThemeTo(systemThemeStyle);
    }
  }, [colorMode, initialColorMode, theme]);

  if (haveFontsLoaded) {
    return (
      <themeContext.Provider
        value={{ theme, colorMode, toggleColorMode } as IThemeContext}
      >
        {children}
      </themeContext.Provider>
    );
  }

  return <AppLoading />;
};
