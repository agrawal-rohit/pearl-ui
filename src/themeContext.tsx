import React, { useEffect, useState, useContext, createContext } from "react";
import { useColorScheme } from "react-native";
import { ThemeProvider as RestyleThemeProvider } from "@shopify/restyle";
import { baseTheme, darkTheme, Theme } from "./theme";
import { ReactElement } from "react-color/node_modules/@types/react";

type ThemeType = "light" | "dark" | "system";

interface IThemeContext {
  themeMode: ThemeType;
  toggleTheme(theme: string): void;
}

interface ThemeProviderProps {
  defaultMode?: ThemeType;
  light?: Theme;
  dark?: Theme;
}

export const themeContext = createContext({} as IThemeContext);

export const ThemeProvider: React.FC<ThemeProviderProps> = ({
  defaultMode = "light",
  light = baseTheme,
  dark = darkTheme,
  children,
}) => {
  const [themeMode, setThemeMode] = useState<ThemeType>(defaultMode);
  const systemThemeStyle = useColorScheme() as ThemeType;

  const toggleTheme = () => {
    if (themeMode === "light") {
      setThemeMode("dark");
    } else {
      setThemeMode("light");
    }
  };

  useEffect(() => {
    // Set theme
    if (defaultMode === "system") {
      setThemeMode(systemThemeStyle);
    }
  }, []);

  return (
    <themeContext.Provider value={{ themeMode, toggleTheme } as IThemeContext}>
      <RestyleThemeProvider theme={themeMode === "light" ? light : dark}>
        {children}
      </RestyleThemeProvider>
    </themeContext.Provider>
  );
};

export const useTheme = () => useContext(themeContext);
