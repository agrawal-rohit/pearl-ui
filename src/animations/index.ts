import { useAnimationState } from "../hooks/useAnimationState";

/**
 * Custom hook for creating a fade animation state.
 * This hook uses the useAnimationState hook to create an animation state that transitions from an opacity of 0 to an opacity of 1.
 * This creates a fade-in effect.
 *
 * @returns An animation state that creates a fade-in effect.
 */
export const useFade = () => {
  // Define the initial and final states for the fade animation.
  // The animation will start with an opacity of 0 (completely transparent) and end with an opacity of 1 (completely opaque).
  const fadeAnimationState = useAnimationState({
    from: {
      opacity: 0,
    },
    to: {
      opacity: 1,
    },
  });

  // Return the fade animation state.
  return fadeAnimationState;
};

/**
 * Custom hook for creating a scale and fade animation state.
 * This hook uses the useAnimationState hook to create an animation state that transitions from a scale of 0.5 and opacity of 0 to a scale of 1 and opacity of 1.
 * This creates a combined effect of scaling and fading in.
 *
 * @returns An animation state that creates a scale and fade-in effect.
 */
export const useScaleFade = () => {
  // Define the initial and final states for the scale and fade animation.
  // The animation will start with a scale of 0.5 (half the original size) and opacity of 0 (completely transparent)
  // and end with a scale of 1 (original size) and opacity of 1 (completely opaque).
  const scaleFadeAnimationState = useAnimationState({
    from: {
      scale: 0.9,
      opacity: 0,
    },
    to: {
      scale: 1,
      opacity: 1,
    },
  });

  // Return the scale and fade animation state.
  return scaleFadeAnimationState;
};

/**
 * Custom hook for creating a slide animation state.
 * This hook uses the useAnimationState hook to create an animation state that transitions from a translateX of -100 to a translateX of 0.
 * This creates a slide-in effect.
 *
 * @returns An animation state that creates a slide-in effect.
 */
export const useSlide = () => {
  // Define the initial and final states for the slide animation.
  // The animation will start with a translateX of -100 (completely off-screen to the left) and end with a translateX of 0 (original position).
  const slideAnimationState = useAnimationState({
    from: {
      translateY: -100,
    },
    to: {
      translateY: 0,
    },
  });

  // Return the slide animation state.
  return slideAnimationState;
};

/**
 * Custom hook for creating a slide and fade animation state.
 * This hook uses the useAnimationState hook to create an animation state that transitions from a translateX of -100 and opacity of 0 to a translateX of 0 and opacity of 1.
 * This creates a combined effect of sliding and fading in.
 *
 * @returns An animation state that creates a slide and fade-in effect.
 */
export const useSlideFade = () => {
  // Define the initial and final states for the slide and fade animation.
  // The animation will start with a translateX of -100 (completely off-screen to the left) and opacity of 0 (completely transparent)
  // and end with a translateX of 0 (original position) and opacity of 1 (completely opaque).
  const slideFadeAnimationState = useAnimationState({
    from: {
      opacity: 0,
      translateY: 10,
    },
    to: {
      opacity: 1,
      translateY: 0,
    },
  });

  // Return the slide and fade animation state.
  return slideFadeAnimationState;
};

/**
 * Custom hook for creating a collapse animation state.
 * This hook uses the useAnimationState hook to create an animation state that transitions from a scaleY of 1 to a scaleY of 0.
 * This creates a collapse effect.
 *
 * @returns An animation state that creates a collapse effect.
 */
export const useCollapse = () => {
  // Define the initial and final states for the collapse animation.
  // The animation will start with a scaleY of 1 (original size) and end with a scaleY of 0 (collapsed).
  const collapseAnimationState = useAnimationState({
    from: {
      opacity: 0,
      height: 0,
    },
    to: {
      opacity: 1,
      height: 100,
    },
  });

  // Return the collapse animation state.
  return collapseAnimationState;
};
