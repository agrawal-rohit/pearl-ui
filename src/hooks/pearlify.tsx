import {
  ComponentTypeProps,
  FinalPearlTheme,
  PearlComponent,
  StyleFunctionContainer,
} from "../theme/src/types";
import React from "react";
import { useAtomicComponentConfig } from "./useAtomicComponentConfig";
import { boxStyleFunctions, BoxStyleProps } from "../theme/src/styleFunctions";
import { useStyledProps } from "./useStyledProps";
import { useMolecularComponentConfig } from "./useMolecularComponentConfig";
import { motify } from "moti";
import { useMotiWithStyleProps } from "./useMotiWithStyleProps";

interface PearlifyConfig {
  componentName: keyof FinalPearlTheme["components"];
  type: "basic" | "atom" | "molecule";
  animatable?: boolean;
}

/**
 * Hook to convert a third-party component to a Pearl atom component that can be configured using style props, an atomic component configuration or a molecular component configuration.
 * @param Component The base component on which the pearlification has to be applied
 * @param config A configuration object that adds metadata to a component that's required by Pearl UI
 * @param styleFunctions The set of style functions that would define the style props that should be supported by the pearlified component
 * @returns
 */
export function pearlify<
  ComponentProps,
  ComponentType extends "basic" | "atom" | "molecule" = "basic",
  StyleProps = BoxStyleProps
>(
  Component: any,
  config: PearlifyConfig = {
    componentName: "None",
    type: "basic",
    animatable: true,
  },
  styleFunctions: StyleFunctionContainer[] = boxStyleFunctions
) {
  let FinalComponent: any;
  if (config.animatable && config.type !== "molecule") {
    // Convert the provided component to a Class component
    class ConvertedClassComponent extends React.Component {
      static displayName = `PearlMoti${Component.name}` || `NoName`;

      render() {
        const { children, ...props } = this.props;

        return (
          <Component ref={(props as any).forwardedRef} {...props}>
            {children}
          </Component>
        );
      }
    }

    FinalComponent = motify(ConvertedClassComponent)();
  } else {
    FinalComponent = Component;
  }

  return React.forwardRef(
    (
      {
        children,
        ...rest
      }: PearlComponent<ComponentProps, StyleProps> &
        ComponentTypeProps<typeof config["componentName"], ComponentType> & {
          children?: string | JSX.Element | JSX.Element[] | React.ReactNode;
        },
      ref: any
    ) => {
      let convertedProps;
      if (config.type === "atom") {
        convertedProps = useAtomicComponentConfig(
          config["componentName"],
          rest,
          {
            size: (rest as any).size,
            variant: (rest as any).variant,
          },
          rest.colorScheme,
          styleFunctions
        );

        if (config.animatable)
          convertedProps = useMotiWithStyleProps(
            convertedProps,
            styleFunctions
          );
      } else if (config.type === "molecule") {
        convertedProps = useMolecularComponentConfig(
          config["componentName"],
          rest,
          {
            size: (rest as any).size,
            variant: (rest as any).variant,
          },
          rest.colorScheme,
          styleFunctions
        );
      } else {
        convertedProps = useStyledProps(rest, styleFunctions);

        if (config.animatable)
          convertedProps = useMotiWithStyleProps(
            convertedProps,
            styleFunctions
          );
      }

      return (
        <FinalComponent {...convertedProps} ref={ref}>
          {children}
        </FinalComponent>
      );
    }
  );
}
