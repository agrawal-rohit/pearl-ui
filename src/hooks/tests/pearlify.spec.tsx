import React from "react";
import { render } from "@testing-library/react-native";
import { ThemeProvider } from "../../theme/src/themeContext";
import { pearlify } from "../pearlify";
import { Text, View } from "react-native";
import { extendTheme } from "../../theme/src/base/index";

jest.useFakeTimers();

const newTheme = extendTheme({
  components: {
    TestView: {
      baseStyle: {
        width: 200,
        height: 200,
        backgroundColor: "danger.400",
        boxShadow: "4xl",
      },
    },
    TestText: {
      baseStyle: {
        color: "success.400",
        fontFamily: "body",
        fontWeight: "hairline",
      },
    },
  },
});

const PearlifyViewComponent: React.FC = ({ children, ...props }) => {
  const PearlView = pearlify(View, "TestView");

  return <PearlView {...props}>{children}</PearlView>;
};

const PearlifyTextComponent: React.FC = ({ children, ...props }) => {
  const PearlText = pearlify(Text, "TestText");

  return <PearlText {...props}>{children}</PearlText>;
};

describe("pearlify", () => {
  it("passes the snapshot test for a <View> component", () => {
    const tree = render(
      <ThemeProvider theme={newTheme}>
        <PearlifyViewComponent></PearlifyViewComponent>
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("passes the snapshot test for a <Text> component", () => {
    const tree = render(
      <ThemeProvider theme={newTheme}>
        <PearlifyTextComponent>asdasd</PearlifyTextComponent>
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
