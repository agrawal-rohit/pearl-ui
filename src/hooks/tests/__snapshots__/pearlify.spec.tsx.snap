// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pearl/atoms passes the snapshot test for a <Text> component 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    color="success.400"
    fontFamily="body"
    fontWeight="hairline"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    asdasd
  </Text>
</View>
`;

exports[`pearl/atoms passes the snapshot test for a <View> component 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    box1={
      Object {
        "backgroundColor": "success.500",
        "height": 50,
        "width": 50,
      }
    }
    root={
      Object {
        "backgroundColor": "danger.400",
        "boxShadow": "4xl",
        "height": 200,
        "p": "4xl",
        "width": 200,
      }
    }
    style={
      Object {
        "backgroundColor": "#FF8371",
        "display": "flex",
        "elevation": 28,
        "height": 200,
        "shadowColor": "#1A2138",
        "shadowOffset": Object {
          "height": 28,
          "width": 0,
        },
        "shadowOpacity": 0.16,
        "shadowRadius": 28,
        "width": 200,
      }
    }
  />
</View>
`;

exports[`pearl/basic passes the snapshot test for a <Text> component 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "display": "flex",
      }
    }
  >
    asdasd
  </Text>
</View>
`;

exports[`pearl/basic passes the snapshot test for a <View> component 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
      }
    }
  />
</View>
`;

exports[`pearl/molecules passes the snapshot test for a <View> component 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    animatedStyle={
      Object {
        "value": Object {
          "transform": Array [],
        },
      }
    }
    collapsable={false}
    style={
      Object {
        "display": "flex",
        "transform": Array [],
      }
    }
  >
    <View
      animatedStyle={
        Object {
          "value": Object {
            "transform": Array [],
          },
        }
      }
      collapsable={false}
      style={
        Object {
          "display": "flex",
          "transform": Array [],
        }
      }
    />
    <View
      animatedStyle={
        Object {
          "value": Object {
            "transform": Array [],
          },
        }
      }
      collapsable={false}
      style={
        Object {
          "display": "flex",
          "transform": Array [],
        }
      }
    />
  </View>
</View>
`;
